src/shared/Core/ApplicationBase.cs

```diff
@@ -75,6 +77,31 @@ public Task<int> RunAsync(string[] args)
                 Context.Trace.WriteLine("Tracing of secrets is enabled. Trace output may contain sensitive information.");
             }
 
+            // Enable TRACE2 tracing
+            if (Context.Settings.GetTrace2TracingEnabled(out string trace2Value))
+            {
+                var connectionPath = PlatformUtils.IsWindows() ? trace2Value.TrimUntilLastIndexOf("/") :
+                    trace2Value.TrimUntilIndexOf(":");
+
+                if (!string.IsNullOrEmpty(connectionPath))
+                {
+                    Context.Trace2.AddListener(new CollectorWriter(
+                            () => new NamedPipeClientStream(".", connectionPath, PipeDirection.Out,
+                                PipeOptions.Asynchronous)
+                        )
+                    );
+
+                    if (AssemblyUtils.TryGetAssemblyVersion(out string version))
+                    {
+                        Context.Trace2.WriteVersion(version);
+                    }
+
+                    var argv = new List<string>() { Context.ApplicationPath };
+                    argv.AddRange(args);
+                    Context.Trace2.WriteStart(argv, Context.ApplicationTimer.ElapsedMilliseconds);
```

Comment:
We've really lost the actual `argv` here since we don't pass it down beyond `Main`.

I wonder if the we need to pass these `Trace2` vars to the `CommandContext` inside of `Main`, along with the constructed timer.

Reply:
you could also fake it here.  just pass something like ["gcm", "store"].
I don't care so much about the pathname to the exe or the other helper args.
we just need to be able to distinguish the different operations.

if you can get the real command line, great, but it's not super critical.